support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutFormGifting.tsx

```diff
@@ -256,44 +232,35 @@ function WeeklyCheckoutFormGifting(props: PropTypes) {
 							id="title"
 							label="Title"
 							optional
-							value={props.titleGiftRecipient}
+							value={props.titleGiftRecipient ?? undefined}
 							onChange={(e) => props.setTitleGift(e.target.value)}
 						>
 							<OptionForSelect>Select a title</OptionForSelect>
 							{options(titles)}
 						</Select>
 						<PersonalDetailsGift
-							firstNameGiftRecipient={props.firstNameGiftRecipient || ''}
+							firstNameGiftRecipient={props.firstNameGiftRecipient ?? ''}
 							setFirstNameGift={props.setFirstNameGift}
-							lastNameGiftRecipient={props.lastNameGiftRecipient || ''}
+							lastNameGiftRecipient={props.lastNameGiftRecipient ?? ''}
 							setLastNameGift={props.setLastNameGift}
-							emailGiftRecipient={props.emailGiftRecipient || ''}
+							emailGiftRecipient={props.emailGiftRecipient ?? ''}
 							setEmailGift={props.setEmailGift}
-							formErrors={
-								props.formErrors as any as Array<
-									FormError<PersonalDetailsFormField>
-								>
-							}
+							formErrors={props.formErrors}
 						/>
 					</FormSection>
 					<FormSection title="Gift delivery date">
 						<Rows>
 							<RadioGroup
 								id="startDate"
-								error={firstError('startDate', props.formErrors)}
-								legend="Gift delivery date"
+								name="startDate"
+								error={firstError('startDate', props.formErrors) as string}
+								label="Gift delivery date"
 							>
 								{days.map((day) => {
 									const [userDate, machineDate] = [
 										formatUserDate(day),
 										formatMachineDate(day),
 									];
-									const hideDate = machineDate.endsWith('-12-25');
-
-									// Don't render input if Christmas day
```

Comment:
no longer needed?

Reply:
It was a short-term fix to address 25th December being on a Friday and therefore listed as an eligible delivery date in 2020. The next time 25th December is on a Friday is 2026, hopefully by then we'll have implemented a better solution! There is a ticket to use the fulfillment date calculator developed by Retention, card to do that is here: https://trello.com/c/bovZtRnp/151-utilise-fulfilment-date-calculator-to-centralise-logic-for-fulfilment-dates.